pennylane/gradients/parameter_shift.py

```diff
@@ -280,23 +282,68 @@ def processing_fn(results):
                 res = fn(res)
 
             # compute the linear combination of results and coefficients
-            res = qml.math.stack(res)
-            g = qml.math.tensordot(res, qml.math.convert_like(coeffs, res), [[0], [0]])
-
-            if unshifted_coeff is not None:
-                # add the unshifted term
-                g = g + unshifted_coeff * r0
+            if qml.active_return():
+
+                multi_measure = len(tape.measurements) > 1
+                if not multi_measure:
+
+                    res = qml.math.stack(res)
+                    g = qml.math.tensordot(res, qml.math.convert_like(coeffs, res), [[0], [0]])
+
+                    if unshifted_coeff is not None:
+                        # add the unshifted term
+                        g = g + unshifted_coeff * r0
+                else:
+                    # TODO: there's a bug likely from here for the multi-var case

```

Comment:
Do you mean multi expval? Here we are in `expval_param_shift`. Or maybe I don't understand how it is related to other measurements here.